specs/AdditionalAllowedFrameAncestors.md

```diff
@@ -0,0 +1,234 @@
+  {
+      m_hostingSite = false;
+  }
+
+  // Using FrameNavigationStarting event instead of NavigationStarting event of CoreWebViewFrame
+  // to cover all possible nested iframes inside the hosted site as CoreWebViewFrame
+  // object currently only support first level iframes in the top page.
+  private void CoreWebView2_FrameNavigationStarting(CoreWebView2 sender, Microsoft.Web.WebView2.Core.CoreWebView2NavigationStartingEventArgs args)
+  {
+      if (IsTargetSite(args.Uri))
+      {
+          args.AdditionalAllowedFrameAncestors = myTrustedSite;
+      }
+  }
+  private void HandleHostedSites()
+  {
+      webView.FrameCreated += CoreWebView2_FrameCreated;
+      webView.FrameNavigationStarting += CoreWebView2_FrameNavigationStarting;
+  }
+```
+
+# API Details
+## Win32 C++
+```
+interface ICoreWebView2NavigationStartingEventArgs_2 : ICoreWebView2NavigationStartingEventArgs
+{
+
+  /// Get additional allowed frame ancestors set by the host app.
+  [propget] HRESULT AdditionalAllowedFrameAncestors([out, retval] LPWSTR* value);
+
+  /// The host may set this property to allow a frame to be hosted by certain additional ancestors besides what is allowed by
+  /// http header [X-Frame-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options)
+  /// and [Content-Security-Policy frame-ancestors directive](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors).
+  /// If set, a frame ancestor is allowed if it is allowed by the additional allowed frame ancestoers or original http header from the site.
+  /// Whether an ancestor is allowed by the additional allowed frame ancestoers is done the same way as if the site provided
+  /// it as the source list of the Content-Security-Policy frame-ancestors directive.
+  /// This property gives the app the ability to use iframe to host sites that otherwise
+  /// could not be hosted in an iframe in trusted app pages.
+  /// This could potentially subject the hosted sites to [Clickjacking](https://en.wikipedia.org/wiki/Clickjacking)
+  /// attack from the code running in the hosting web page. Therefore, you should only
+  /// set this property with origins of fully trusted hosting page and any intermediate iframes.
+  /// Whenever possible, you should use the list of specific origins of the top and intermediate
+  /// frames instead of wildcard characters for this property.
+  /// This API is to provide limited support for app scenarios that used to be supported by
+  /// `<webview>` element in other solutions like JavaScript UWP apps and Electron.
+  /// You should limit the usage of this property to trusted pages, and if possible, to specific iframe and
+  /// specific navigation target url, by checking the `Source` of the WebView2, the `Name`
+  /// of the ICoreWebView2Frame and `Uri` of the event args.
+  /// This property is ignored for top level document navigation.
+  ///
```

Comment:
It would be nice to include some examples to demonstrate the expected sort of thing of setting a single host-source, and a complicated example demonstrating setting multiple of all the supported source kinds.

```suggestion
  ///
  /// For example, to additionally allow `https://example.com` as a frame ancestor you would specify:
  /// `https://example.com`
  ///
  /// Or to additionally allow the web content's own origin, any subdomain of example.com, and all blob URIs you would specify:
  /// `'self' https://*.example.com blob:`
```

Reply:
Don't want to demonstrate * or scheme entries, but we do want an example with more than one host entry. Please update.